### PR TITLE
Reimplement search and advanced-search features in React

### PR DESCRIPTION
**Pull request by syucream**
It reimplement search and advanced-search features in React, like this.

# Advanced search

image

# Search result

image

# Demo video

https://user-images.githubusercontent.com/191684/135706754-900a8e44-4acf-40a0-b72f-47521091b85e.mov

### CODE REVIEW

File: frontend/src/utils/AironeAPIClient.js [hunk 1]
```diff
@@ -53,6 +57,28 @@ export function getEntries(entityId) {
...
 }
 
+// FIXME it should be better to implement a new internal API than this
+export function searchEntries(
+  entityIds = [],
+  entryName = "",
+  attrInfo = [],
+  entryLimit = 99999
+) {
+  return fetch(`/api/v1/entry/search`, {
```
Comment (takano32) [inline frontend/src/utils/AironeAPIClient.js:81]: Is it good at using `/api/v1` API in Reactize code?

**Review by userlocalhost — COMMENTED** [review]
It's better to be able to narrow down selecting Entities and Attributes for improving usability when there are many Entities and Attributes are registered, like original UI is.

## Advanced search settings page of new UI
スクリーンショット 2021-08-23 11 14 25

## Advanced search settings page of original UI
スクリーンショット 2021-08-23 11 17 08

File: frontend/src/utils/AironeAPIClient.js (hunk 1 above)
Comment (syucream) [reply frontend/src/utils/AironeAPIClient.js:81]: Yes it is! I just focus to realize fundamental UI components on this PR, so I want to aim it on another issue. https://github.com/dmm-com/airone/issues/217

**Comment by syucream** [comment]
@userlocalhost So lets have a text field to accept filter keyword like the UI! How about style of the query(entities, attributes) selectors? My idea is:

- Transfer List (similar to current UI) https://material-ui.com/components/transfer-list/
- Checklist (current implementation) https://material-ui.com/components/lists/#checkbox
- Multiple select https://material-ui.com/components/selects/#multiple-select
- etc

Any preferred way?

**Review by takano32 — APPROVED** [review]
LGTM

**Comment by hinashi** [comment]
I can't following things
- filter candidates to search at text box
- select multiple candidates at once (e.g. shift + select)

image

And I need following feature
- loading view at waiting API response

**Comment by syucream** [comment]
@userlocalhost @hinashi I've implemented it with multi-selectable checkbox and name filter. How about this?

**Comment by userlocalhost** [comment]
I'm afraid to say that I want to filter Entities and Attributes with case insensitive search as ordinal UI.
スクリーンショット 2021-10-05 15 07 40
スクリーンショット 2021-10-05 15 07 55

**Comment by userlocalhost** [comment]
I need feature to be able to select all Attributes to filter at once.
スクリーンショット 2021-10-05 15 27 45

**Pull request merged**